src/features/query-bar/components/filter/Condition.js

```diff
@@ -58,88 +56,84 @@ const Condition = ({
     };
 
-    const updateSelectedField = (option: SelectOptionType, fieldType?: string) => {
-        const conditionIndex = index;
-        const value = option.value;
-        const valueType = option.type;
-        const fieldId = option.fieldId || condition.fieldId;
+    const updateSelectedField = (option: OptionType, fieldType?: string) => {
+        const { displayText, type, value } = option;
 
-        const displayText = option.displayText;
         let displayTextType = '';
         let keyType = '';
 
-        if (fieldType === ATTRIBUTE) {
-            displayTextType = COLUMN_DISPLAY_TEXT;
-            keyType = COLUMN_KEY;
-        } else if (fieldType === OPERATOR) {
-            displayTextType = OPERATOR_DISPLAY_TEXT;
-            keyType = OPERATOR_KEY;
-        } else if (fieldType === VALUE) {
-            displayTextType = VALUE_DISPLAY_TEXT;
-            keyType = VALUE_KEY;
+        switch (fieldType) {
+            case COLUMN:
+                displayTextType = COLUMN_DISPLAY_TEXT;
+                keyType = COLUMN_KEY;
+                break;
+            case OPERATOR:
+                displayTextType = OPERATOR_DISPLAY_TEXT;
+                keyType = OPERATOR_KEY;
+                break;
+            case VALUE:
+                displayTextType = VALUE_DISPLAY_TEXT;
+                keyType = VALUE_KEY;
+                break;
+            default:
+                throw new Error('invalid input');
         }
 
-        update(conditionIndex, condition, displayText, displayTextType, fieldId, value, keyType, valueType);
+        onFieldChange(index, condition, displayText, displayTextType, value, keyType, type);
     };
 
     const getFormattedOptions = (options: Array<Object>): any[] => {
         return options.map(option => {
+            const { displayName, type } = option;
             return {
-                displayText: option.displayName,
-                fieldId: option.id,
-                type: option.type,
-                value: option.displayName,
+                displayText: displayName,
+                type,
+                value: displayName,
             };
         });
     };
 
-    const getOperatorsForAttribute = () => {
+    const getColumnOperators = () => {
         const { valueType } = condition;
         if (valueType === '') {
             return [];
         }
-        return OPERATORS_FOR_ATTRIBUTE[valueType];
+        return COLUMN_OPERATORS[valueType];
     };
 
-    const getValuesForAttribute = () => {
-        const { fieldId } = condition;
-        const field =
-            columns &&
-            columns.find(column => {
-                return column.id === fieldId;
-            });
-
-        if (field && field.options) {
-            const fieldOptions = field.options.map((option, optionIndex) => {
+    const getColumnOptions = () => {
+        const { columnKey } = condition;
+        const column = columns && columns.find(c => c.displayName === columnKey);
```

Comment:
```c.displayName === columnKey```

Comparing a label with a key doesn't seem right.

Reply:
Will be addressing this in a separate PR